app/controllers/agents/agents_controller.rb

```diff
@@ -12,7 +12,7 @@ def show
     authorize(@agent)
     @status = params[:status]
     @organisation = current_organisation
-    @date = params[:date] ? Date.parse(params[:date]) : nil
+    @date = !params[:date].blank? ? Date.parse(params[:date]) : nil
```

Comment:
`.present?` plus explicite ?

https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html